zerver/lib/default_streams.py

```diff
@@ -0,0 +1,33 @@
+from typing import List, Set
+
+from zerver.lib.types import APIStreamDict
+from zerver.models import DefaultStream, Stream
+
+
+def get_slim_realm_default_streams(realm_id: int) -> List[Stream]:
+    # We really want this query to be simple and just get "thin" Stream objects
+    # in one round trip.
+    #
+    # The above is enforced by at least three tests that verify query counts,
+    # and test_query_count in test_subs.py makes sure that the query itself is
+    # not like 11000 bytes, which is what you would get if you did
+    # select_related().  Please avoid select_related() with streams, because
+    # you will get all kinds of query bloat related to Realm and UserGroup.
+    streams = Stream.objects.filter(
+        id__in=DefaultStream.objects.filter(realm_id=realm_id).values_list("stream_id")
+    )
+    return list(streams)
```

Comment:
Thinking about this again, this can be simplified to `Stream.objects.filter(defaultstream__realm_id=realm_id)`.

The “avoid `select_related()`” advice is too broad and may mislead people into unnecessarily doing multiple queries. The right advice is “avoid `select_related()` with no arguments”.

Reply:
I thought that was sort of implied by saying `select_related()` instead of `select_related`, but I'll clarify the comment and use `Stream.objects.filter(defaultstream__realm_id=realm_id)` and re-push.